feature(standalone-toasts): create toasts outside of react components

TimKolberger opened the pull request:
As mentioned in #1693 application code outside of the react context should be able to create a new toast.
E.g. in API abstractions or Vanilla JS files it can simplify the code.

Fix #1693

## Example 

```
import { createStandaloneToast } from "@chakra/core"

const toast = createStandaloneToast()
// const customToast = createStandaloneToast({ theme: customTheme })

const fetchUsers = async () => {
  await fetch() // ...
  toast({
    title: "Yay",
    status: "success"
  })
}
```

with-heart commented on packages/toast/test/standalone-toast.test.tsx:
```diff
@@ -0,0 +1,31 @@
+import React, { useEffect } from "react"
+import { render } from "@chakra-ui/test-utils"
+import { createStandaloneToast } from "../src"
+import theme from "@chakra-ui/theme"
```
Make sure to add `@chakra-ui/theme` to `devDependencies` in `packages/toast/package.json`

TimKolberger replied:
✅  done

with-heart commented on packages/toast/test/standalone-toast.test.tsx:
```diff
@@ -0,0 +1,31 @@
+
+  const { findAllByText } = render(<ToastContainer />)
+
+  const allByTitle = await findAllByText(title)
```
Can you switch this to use `screen.findAllByText` rather than destructuring from `render`?

TimKolberger replied:
✅  done

with-heart commented on packages/toast/test/standalone-toast.test.tsx:
```diff
@@ -0,0 +1,31 @@
+
+  const ToastContainer = () => {
+    useEffect(() => {
+      toast({
```
Very cool!

TimKolberger replied:
❤️

jmiazga commented:
Is there a benefit for keeping the useToast hook?

TimKolberger commented:
@jmiazga Yes, the hook `const toast = useToast()` will get the current chakra theme from the React context. 
This way it does not depend on the developer to pass the theme explicitly.

With the standalone version you need to pass your theme like: `const toast = createStandaloneToast(theme)`

trompx commented:
I'm trying to use Toast with apollo-link-error. I'm not sure if this will work but looks like it. Anything I can do to help?

TimKolberger commented:
I rebased the branch. The vercel Deployment should be working now.

TimKolberger commented:
Is there anything I can do to bring this PR forward?
Otherwise I will just keep rebasing it with `develop` 😅

TimKolberger commented:
Due some changes in the useToast hook I updated the PR accordingly.
I refactored the duplicated render function into one and added default values for the object parameter of `createStandaloneToast`.

@with-heart Your last code review is outdated due those changes.

Will this PR be part of v1?

ljosberinn commented:
@TimKolberger sorry for the delay here - we're packed with work at ... work atm. I'm sure Mark will find time soon! 🤗

TimKolberger commented:
Sure - no pressure intended! =) 
I am a bit confused about the Github UI which says "Changes requested", but I those are outdated.
Just want to make sure we are not waiting on each other :D

ljosberinn commented:
He might have to manually accept the changes and until then its pending; I can't update anything here myself either :/
